File: backend/instrumentos/src/main/java/com/utn/instrumentos/entities/Instrumento.java
package com.utn.instrumentos.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
public class Instrumento {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idinstrumento;

    private String instrumento;
    private String marca;
    private String modelo;
    private String imagen;
    private Double precio;
    private String costoenvio;
    private Integer cantidadvendida;
    private String descripcion;
    private int cantidadEnPedido;

    @ManyToOne
    @JoinColumn(name = "idcategoria")
    private Categoria categoria;

}

File: frontend/src/types/Instrumento.ts
import { Categoria } from "./Categoria";

export interface Instrumento {
    idinstrumento: number;
    instrumento: string;
    marca: string;
    modelo: string;
    imagen: string;
    precio: number;
    costoenvio: string;
    cantidadvendida: number;
    descripcion: string;
    idcategoria: Categoria;
    cantidadEnCarrito: number;
}